server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt

```diff
@@ -50,35 +45,38 @@ object MangaList {
 
     fun MangasPage.insertOrGet(sourceId: Long): List<Int> {
         return transaction {
-            mangas.map { manga ->
-                val mangaEntry =
-                    MangaTable.select {
-                        (MangaTable.url eq manga.url) and (MangaTable.sourceReference eq sourceId)
-                    }.firstOrNull()
-                if (mangaEntry == null) { // create manga entry
-                    val mangaId =
-                        MangaTable.insertAndGetId {
-                            it[url] = manga.url
-                            it[title] = manga.title
+            val existingMangaUrlsToId =
+                MangaTable.slice(MangaTable.url, MangaTable.id).select {
+                    MangaTable.url inList mangas.map { it.url }
```

Comment:
Source id needs to be taken into account.
```suggestion
                    MangaTable.sourceReference eq sourceId and (MangaTable.url inList mangas.map { it.url })
```

Reply:
fixed